packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -298,15 +298,19 @@ export class Combobox
       return;
     }
 
-    this.setInactiveIfNotContained(event);
+    const composedPath = event.composedPath();
+    if (!this.open || composedPath.includes(this.el) || composedPath.includes(this.referenceEl)) {
```

Comment:
Can you add a test that explicitly checks that the custom input value doesn't clear when the compose path includes this.el?

Reply:
we do have a test for this one currently. Any item selection test covers this usecase.